challenge_8/python/mjuiuc/solution.py

```diff
@@ -0,0 +1,23 @@
+from listNode import listNode
+
+
+def solution(head):
+	lop = dict()
+	newHead = listNode()
+	temp = head
+	temp2 = newHead
+	# copy the list first
+	while temp != None:
+		temp2.val = temp.val
+		lop[temp2.val] = temp2
```

Comment:
Oh yea you're right. Hmmmm I wonder if python has a data structure that's similar to an unordered multi map in c++.